diff --git a/node_modules/@toptal/davinci-engine/src/utils/publish-packages/publish-monorepo-packages.js b/node_modules/@toptal/davinci-engine/src/utils/publish-packages/publish-monorepo-packages.js
index d357ec5..b8c7a8e 100644
--- a/node_modules/@toptal/davinci-engine/src/utils/publish-packages/publish-monorepo-packages.js
+++ b/node_modules/@toptal/davinci-engine/src/utils/publish-packages/publish-monorepo-packages.js
@@ -34,6 +34,7 @@ export const publishMonorepoPackages = ({
   const result = cliShared.runSync(publishCommand, [], {
     shell: true,
     stdio: 'pipe',
+    all: true,
   })
 
   // prints the result