diff --git a/esm/store/ReactStore.d.ts b/esm/store/ReactStore.d.ts
index 0000000..1111111 100644
--- a/esm/store/ReactStore.d.ts
+++ b/esm/store/ReactStore.d.ts
@@ -70,7 +70,7 @@
    *
    * @param key Key of the state to set.
    */
-  useStateSetter<const Key extends keyof State, Value extends State[Key]>(key: keyof State): (v: Value) => void;
+  useStateSetter<Key extends keyof State, Value extends State[Key]>(key: keyof State): (v: Value) => void;
   /**
    * Observes changes derived from the store's selectors and calls the listener when the selected value changes.
    *
diff --git a/esm/store/createSelector.d.ts b/esm/store/createSelector.d.ts
index 0000000..1111111 100644
--- a/esm/store/createSelector.d.ts
+++ b/esm/store/createSelector.d.ts
@@ -9,7 +9,7 @@
  * as memoization relies on the Function.length property, which does not account for optional or default parameters.
  */
 type NoOptionalParams<F extends Fn> = Parameters<F> extends Required<Parameters<F>> ? F : 'Combiner cannot have optional or default parameters because memoization relies on Function.length';
-export type CreateSelectorFunction = <const Args extends any[], const Selectors extends ReadonlyArray<Selector<any>>, const Combiner extends (...args: readonly [...ReturnTypes<Selectors>, ...Args]) => any>(...items: [...Selectors, NoOptionalParams<Combiner>]) => (...args: Selectors['length'] extends 0 ? MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>> : [StateFromSelectorList<Selectors>, ...MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>>]) => ReturnType<Combiner>;
+export type CreateSelectorFunction = <Args extends any[], Selectors extends ReadonlyArray<Selector<any>>, Combiner extends (...args: readonly [...ReturnTypes<Selectors>, ...Args]) => any>(...items: [...Selectors, NoOptionalParams<Combiner>]) => (...args: Selectors['length'] extends 0 ? MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>> : [StateFromSelectorList<Selectors>, ...MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>>]) => ReturnType<Combiner>;
 type StateFromSelectorList<Selectors extends readonly any[]> = Selectors extends [f: infer F, ...other: infer R] ? StateFromSelector<F> extends StateFromSelectorList<R> ? StateFromSelector<F> : StateFromSelectorList<R> : {};
 type StateFromSelector<T> = T extends ((first: infer F, ...args: any[]) => any) ? F : never;
 type DropFirst<T> = T extends [any, ...infer Xs] ? Xs : [];
diff --git a/store/ReactStore.d.ts b/store/ReactStore.d.ts
index 0000000..1111111 100644
--- a/store/ReactStore.d.ts
+++ b/store/ReactStore.d.ts
@@ -70,7 +70,7 @@
    *
    * @param key Key of the state to set.
    */
-  useStateSetter<const Key extends keyof State, Value extends State[Key]>(key: keyof State): (v: Value) => void;
+  useStateSetter<Key extends keyof State, Value extends State[Key]>(key: keyof State): (v: Value) => void;
   /**
    * Observes changes derived from the store's selectors and calls the listener when the selected value changes.
    *
diff --git a/store/createSelector.d.ts b/store/createSelector.d.ts
index 0000000..1111111 100644
--- a/store/createSelector.d.ts
+++ b/store/createSelector.d.ts
@@ -9,7 +9,7 @@
  * as memoization relies on the Function.length property, which does not account for optional or default parameters.
  */
 type NoOptionalParams<F extends Fn> = Parameters<F> extends Required<Parameters<F>> ? F : 'Combiner cannot have optional or default parameters because memoization relies on Function.length';
-export type CreateSelectorFunction = <const Args extends any[], const Selectors extends ReadonlyArray<Selector<any>>, const Combiner extends (...args: readonly [...ReturnTypes<Selectors>, ...Args]) => any>(...items: [...Selectors, NoOptionalParams<Combiner>]) => (...args: Selectors['length'] extends 0 ? MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>> : [StateFromSelectorList<Selectors>, ...MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>>]) => ReturnType<Combiner>;
+export type CreateSelectorFunction = <Args extends any[], Selectors extends ReadonlyArray<Selector<any>>, Combiner extends (...args: readonly [...ReturnTypes<Selectors>, ...Args]) => any>(...items: [...Selectors, NoOptionalParams<Combiner>]) => (...args: Selectors['length'] extends 0 ? MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>> : [StateFromSelectorList<Selectors>, ...MergeParams<ReturnTypes<Selectors>, Parameters<Combiner>>]) => ReturnType<Combiner>;
 type StateFromSelectorList<Selectors extends readonly any[]> = Selectors extends [f: infer F, ...other: infer R] ? StateFromSelector<F> extends StateFromSelectorList<R> ? StateFromSelector<F> : StateFromSelectorList<R> : {};
 type StateFromSelector<T> = T extends ((first: infer F, ...args: any[]) => any) ? F : never;
 type DropFirst<T> = T extends [any, ...infer Xs] ? Xs : [];
